variables.tf

```diff
@@ -1,9 +1,444 @@
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API key which will enable billing exports"
+  sensitive   = true
+}
+
+variable "cloudability_api_key" {
+  type        = string
+  description = "API Key for Apptio Cloudability"
```

Comment:
hmm, at the module level it would perform all of the infrastructure creation on the IBM Cloud side and then only if this api key was set would it add the account to cloudability. Since there is no way to have an conditional provider in terraform I will just make this a required variable then

Reply:
@chrisw-ibm if it is indeed optional, and the provider doesn't give out when its null then thats fine. Just update the description to make it clear what happens when its set VS not set

Reply:
however, for my tests. It ran fine with setting this to null and in this case all of the IBM Cloud infrastructure does get created without onboarding to to cloudability. So I will keep the default as null. But it is set as a required field in the `ibm_catalog.json`